pkg/kubectl/cmd/get.go

```diff
@@ -182,13 +183,22 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 		return cmdutil.UsageError(cmd, usageString)
 	}
 
-	// always show resources when getting by name or filename
+	// determine if args contains "all"
+	for _, a := range args {
+		if a == "all" {
```

Comment:
> Although unrelated to the output type, I was thinking it would be a better user experience overall if all implied that even resources that are hidden by default would also be displayed.

This might be a lot of things, since this is showing both all resource types and hidden resources. 

> Would it be a valid assumption that if all comes before any other "kind", it can be implied that the user did mean "all resources", rather than pod/all? If this is the case, could we only check to see if args[0] == "all", or would that be too fragile?

Maybe we should check there is only 1 argument, and it is all?

```go
if len(args) == 1 && args[0] == "all"
```